relayer/src/main.rs

```diff
@@ -1,3 +1,307 @@
+                                        report.to_string()
+                                )
+                            );
+                            info!("{}", err_msg);
+                            return (
+                                StatusCode::INTERNAL_SERVER_ERROR,
+                                err_msg,
+                                ).into_response()
+                        },
+                    },
+                };
+            };
+
+            // build response json
+            let mut success_msg_json: Map<String, Value> = Map::new();
+            success_msg_json.insert("message".to_string(),
+                                    json!("Successfully relayed and sent transaction."));
+            success_msg_json.insert("Status".to_string(),
+                                    json!(transaction_info.status));
+            success_msg_json.insert("Transaction Outcome Logs".to_string(),
+                                    json!(transaction_info.transaction_outcome.outcome.logs.join("\n")));
+            info!("Success message: {:?}", success_msg_json);
+            let success_msg_str = serde_json::to_string(&success_msg_json).unwrap();
+            success_msg_str.into_response()
+        },
+        Err(e) => {
+            let err_msg = String::from(
+                format!("{}: {:?}",
+                    "Error deserializing payload data object".to_string(),
+                    e.to_string()
+                )
+            );
+            info!("{}", err_msg);
+            (
+                StatusCode::BAD_REQUEST,
+                err_msg,
+            ).into_response()
+        },
+    }
+}
+
+/**
+--------------------------- Testing below here ---------------------------
+*/
+#[cfg(test)]
+fn create_signed_delegate_action(
+    sender_id: String,
+    receiver_id: String,
+    actions: Vec<Action>,
+    nonce: i32,
+    max_block_height: i32
+) -> SignedDelegateAction {
+    let max_block_height: i32 = max_block_height;
+    let public_key: PublicKey = PublicKey::empty(KeyType::ED25519);
+    let signature: Signature = Signature::empty(KeyType::ED25519);
+    SignedDelegateAction {
+        delegate_action: DelegateAction {
+            sender_id: sender_id.parse().unwrap(),
+            receiver_id: receiver_id.parse().unwrap(),
+            actions: actions
+                .iter()
+                .map(|a| NonDelegateAction::try_from(a.clone()).unwrap())
+                .collect(),
+            nonce: nonce as Nonce,
+            max_block_height: max_block_height as BlockHeight,
+            public_key,
+        },
+        signature,
+    }
+}
+
+#[tokio::test]
+async fn test_relay() {   // tests assume testnet in config
+    // Test Transfer Action and a CreateAccount Action
+    let actions = vec![
+        Action::CreateAccount(CreateAccountAction {}),
+        Action::Transfer(TransferAction { deposit: 1 })
+    ];
+    let sender_id: String = String::from("nomnomnom.testnet");
```

Comment:
Could we use different accounts for this test? Using the same account doesn't reflect the change in `signer_id` `receiver_id` changes when we are converting a `SignedDelegateAction` to `DelegateAction`.